src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -88,8 +89,19 @@ private static IServiceCollection AddSignalRServiceCore(this IServiceCollection
             // On .NET Standard 2.0, registering multiple hub protocols with the same name is forbidden.
             services.Replace(ServiceDescriptor.Singleton<IHubProtocol>(sp =>
             {
-                var objectSerializer = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.ObjectSerializer;
-                return objectSerializer != null ? new JsonObjectSerializerHubProtocol(objectSerializer) : new JsonHubProtocol();
+                var serviceManagerOptions = sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value;
+                var objectSerializer = serviceManagerOptions.ObjectSerializer;
+                if (objectSerializer != null)
+                {
+                    return new JsonObjectSerializerHubProtocol(objectSerializer);
+                }
+#pragma warning disable CS0618 // Type or member is obsolete
+                // The default protocol is different for historical reason.
+                return serviceManagerOptions.ServiceTransportType == default ?
```

Comment:
Use the explicit value for readability 
```suggestion
                return serviceManagerOptions.ServiceTransportType == ServiceTransportType.Transient ?
```